apps/analysis_framework/models.py

```diff
@@ -64,21 +61,92 @@ def can_modify(self, user):
         * user is super user, or
         * the framework belongs to a project where the user is admin
         """
-        import project
         return (
             self.created_by == user or
             user.is_superuser or
-            project.models.ProjectMembership.objects.filter(
-                project__in=self.project_set.all(),
+            AnalysisFrameworkMembership.objects.filter(
                 member=user,
-                role__in=project.models.ProjectRole.get_admin_roles(),
+                framework=self,
+                role__can_edit_framework=True
+            ).exists()
+        )
+
+    def can_clone(self, user):
+        return (
+            not self.is_private or
```

Comment:
`or` or `and`?

Reply:
It's `or`. Firstly `not self.is_private` is evaluated which means the framework is public. So the overall statement is `can clone if public or has clone permissions`.